frb_dart/lib/src/ffi/io.dart

```diff
@@ -64,3 +70,17 @@ class FrbOpaqueBase implements Finalizable {
           int size, OpaqueTypeFinalizer finalizer) =>
       finalizer.attach(opaque, ptr, detach: opaque, externalSize: size);
 }
+
+int getOpaquePtr(Uint8List list) {
+  var pointBitLen = 8;
```

Comment:
⚠️ Dart may run on 32bit, so are we sure it is really "8"?

Reply:
IIRC flutter doesn't support x86.

```
final int pointerByteLength = (double.maxFinite.toInt().bitLength + 1) ~/ 8;
final int Function(Uint8List list) getPlatformUsize = pointerByteLength == 8
    ? (Uint8List list) => ByteData.view(list.buffer).getUint64(0)
    : (Uint8List list) => ByteData.view(list.buffer).getUint32(0);
```

Reply:
> final int pointerByteLength = (double.maxFinite.toInt().bitLength + 1) ~/ 8;

Hmm have you tested in 32bit android? IIRC (https://dart.dev/guides/language/numbers) dart int and double are still 64bit even in 32bit machines.

But Rust usize may be 32bit imho.

Another idea is, what about *always* being 64bit. i.e. send a `u64` (or `i64`) from Rust, instead of usize

Reply:
What if
```
late int _pointerByteLength;
// ignore: unnecessary_late
late final _getPlatformUsize = _pointerByteLength == 8
    ? (Uint8List data) => ByteData.view(data.buffer).getUint64(0)
    : (Uint8List data) => ByteData.view(data.buffer).getUint32(0);

int getPlatformUsize(Uint8List data) {
  _pointerByteLength = data.length;
  return _getPlatformUsize(data);
}

Tuple2<int, int> parseOpaquePtrAndSizeFrom(Uint8List data) {
  var pointerLength = data.length ~/ 2;

  var ptrList = List.filled(pointerLength, 0);
  List.copyRange(ptrList, 0, data, 0, pointerLength);

  var sizeList = List.filled(pointerLength, 0);
  List.copyRange(sizeList, 0, data, pointerLength, pointerLength * 2);

  return Tuple2(getPlatformUsize(Uint8List.fromList(ptrList)),
      getPlatformUsize(Uint8List.fromList(sizeList)));
}
```
overhead - overwrite int
versus copying 4 extra bytes for 32bit platforms

Reply:
That looks like some more overhead IMHO.

Given dart int is 64bit even on 32bit machines, I guess sending 64bits (instead of usize) is not a problem.